fastly/pool.go

```diff
@@ -0,0 +1,293 @@
+	MaxConnDefault   uint     `mapstructure:"max_conn_default"`
+	ConnectTimeout   uint     `mapstructure:"connect_timeout"`
+	FirstByteTimeout uint     `mapstructure:"first_byte_timeout"`
+	Quorum           uint     `mapstructure:"quorum"`
+	UseTLS           bool     `mapstructure:"use_tls"`
+	TLSCACert        string   `mapstructure:"tls_ca_cert"`
+	TLSCiphers       string   `mapstructure:"tls_ciphers"`
+	TLSClientKey     string   `mapstructure:"tls_client_key"`
+	TLSClientCert    string   `mapstructure:"tls_client_cert"`
+	TLSSNIHostname   string   `mapstructure:"tls_sni_hostname"`
+	TLSCheckCert     bool     `mapstructure:"tls_check_cert"`
+	TLSCertHostname  string   `mapstructure:"tls_cert_hostname"`
+	MinTLSVersion    string   `mapstructure:"min_tls_version"`
+	MaxTLSVersion    string   `mapstructure:"max_tls_version"`
+	Healthcheck      string   `mapstructure:"healthcheck"`
+	Type             PoolType `mapstructure:"type"`
+	OverrideHost     string   `mapstructure:"override_host"`
+}
+
+// poolsByName is a sortable list of pools.
+type poolsByName []*Pool
+
+// Len, Swap, and Less implement the sortable interface.
+func (s poolsByName) Len() int      { return len(s) }
+func (s poolsByName) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+func (s poolsByName) Less(i, j int) bool {
+	return s[i].Name < s[j].Name
+}
+
+// ListPoolsInput is used as input to the ListPools function.
+type ListPoolsInput struct {
+	// Service is the ID of the service (required).
+	Service string
+
+	// Version is the specific configuration version (required).
+	Version int
+}
+
+// ListPools lists all pools for a particular service and version.
+func (c *Client) ListPools(i *ListPoolsInput) ([]*Pool, error) {
+	if i.Service == "" {
+		return nil, ErrMissingService
+	}
+
+	if i.Version == 0 {
+		return nil, ErrMissingVersion
+	}
+
+	path := fmt.Sprintf("/service/%s/version/%d/pool", i.Service, i.Version)
+	resp, err := c.Get(path, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var ps []*Pool
+	if err := decodeJSON(&ps, resp.Body); err != nil {
+		return nil, err
+	}
+	sort.Stable(poolsByName(ps))
+	return ps, nil
+}
+
+// CreatePoolInput is used as input to the CreatePool function.
+type CreatePoolInput struct {
+	// Service is the ID of the service. Version is the specific configuration
+	// version. Both fields are required.
+	Service string
+	Version int
+
+	// Name of the pool (required).
+	Name string `form:"name"`
+
+	Comment          string       `form:"comment,omitempty"`
+	Shield           string       `form:"shield,omitempty"`
+	RequestCondition string       `form:"request_condition,omitempty"`
+	MaxConnDefault   uint         `form:"max_conn_default,omitempty"`
+	ConnectTimeout   uint         `form:"connect_timeout,omitempty"`
+	FirstByteTimeout uint         `form:"first_byte_timeout,omitempty"`
+	Quorum           uint         `form:"quorum,omitempty"`
+	UseTLS           *Compatibool `form:"use_tls,omitempty"`
```

Comment:
Are you sure this needs to be a `Compatibool`?

Reply:
I'm not sure that this needs to be a `Compatibool`, but the API documentation for the [Pool resource](https://docs.fastly.com/api/dynamicservers#api-section-pool) suggests that this should be a [Compatibool](https://github.com/fastly/go-fastly/blob/e18f1a520c404b08eb1d2c219b9010de494aab47/fastly/fastly.go#L48-L50):

>field: use_tls 	
>type: string 	
>description: Whether or not to use TLS to reach any server in the pool. Optional. Defaults to "0".

so that this is marshalled to a "0/1" rather than "true/false". Let me know if that's not the case.

Reply:
Thanks for checking, I'm happy leaving it as a `Compatibool` 👍